src/utils/tracked-controls.js

```diff
@@ -88,10 +88,20 @@ function findMatchingController (controllers, filterIdExact, filterIdPrefix, fil
   var i;
   var matchingControllerOccurence = 0;
   var targetControllerMatch = filterControllerIndex || 0;
-
+  var filterIdPrefixes;
```

Comment:
This would all be cleaner if you just made a RegExp that matches on the controller.id with a starts with operator "^". This will be jit compiled (faster) and remove a bunch of code.

Reply:
That's a good idea. I always think twice about having RegExps if I can avoid them.

Reply:
If we go with this approach, probably better to just make all the prefixes into regex and use it as such...